src/libs/utils.js

```diff
@@ -772,3 +772,12 @@ export const completedResearcherInfoCheck = (properties) => {
   } = properties;
   return !isNil(institutionId);
 };
+
+export const researcherProfileComplete = (user) => {
+  const {
+    institutionId,
+    selectedSigningOfficialId,
+  } = getPropertyValuesFromUser(user);
+
+  return !isNil(institutionId) && !isNil(selectedSigningOfficialId);
```

Comment:
Is this how we want to check if a profile is complete? Right now, checks that there is a valid institution id and selected sigining official id; in the case that the user suggests a signing official / institution, their profile is still considered incomplete; this seems right to me, but wanted to make sure.

Reply:
thought: is the expected behavior for a user which is a signing official to always skip this ?

Reply:
> Is this how we want to check if a profile is complete? Right now, checks that there is a valid institution id and selected sigining official id; in the case that the user suggests a signing official / institution, their profile is still considered incomplete; this seems right to me, but wanted to make sure.

I talked with @solideoglori about this and we both decided not to worry about a "complete" profile and only have the redirect occur on users that are registering with DUOS for the first time. Depending on the role you may not care about SO or Institution, so nil checks on both won't be enough. At the same time the system allows for multiple roles which just adds complexity.

From a product perspective the odds of someone signing out of the profile page without doing anything is relatively low. If a user does do that, the only features that would require checks are the DAR Form and the SO console. The DAR page already performs that check and provides the user with a message and link to the profile page. For the SO we only assign that role if the user has an institution so we're okay there as well.

**TLDR**: Don't worry about a "complete" profile, just perform the redirect for new users.